library/Icingadb/Widget/ViewModeSwitcher.php

```diff
@@ -25,41 +27,38 @@ class ViewModeSwitcher extends BaseHtmlElement
         'detailed' => 'detailed'
     ];
 
-    /** @var Url */
-    protected $url;
-
     /** @var string */
-    protected $viewModeParam = self::DEFAULT_VIEW_MODE_PARAM;
+    protected $defaultViewMode;
 
-    protected $tag = 'ul';
+    /** @var string */
+    protected $method = 'POST';
 
-    protected $defaultAttributes = ['class' => 'view-mode-switcher'];
+    /** @var callable */
+    protected $protector;
 
-    public function __construct(Url $url)
-    {
-        $this->url = $url;
-    }
+    /** @var string */
+    protected $viewModeParam = self::DEFAULT_VIEW_MODE_PARAM;
 
     /**
-     * Get the base url
+     * Get the default mode
      *
-     * @return Url
+     * @return string
      */
-    public function getUrl()
+    public function getDefaultViewMode()
     {
-        return $this->url;
+        return $this->defaultViewMode ?: static::DEFAULT_VIEW_MODE;
     }
 
     /**
-     * Set the base url
+     * Set the default view mode
      *
-     * @param Url $url
+     * @param String $defaultViewMode
```

Comment:
`string`, lowercase, as type please. Otherwise it may be mis-interpreted as class name.

Reply:
✅